### ISSUE TITLE
DateTime() modify('first day of this month') wrong output

### ISSUE THREAD

**Issue by Cholik** [facebook/hhvm#4681]
Code i used:

```php
$begin = new DateTime('2015-01-20 00:00:00.000000');
$begin->modify('first day of this month');
var_dump($begin);
```

Output in all PHP Versions i tried is: 2015-01-01
HHVM output is: 2015-01-20

**Output for hhvm-3.3.0 - 3.4.1**
```
object(DateTime)#1 (3) {
  ["date"]=>
  string(26) "2015-01-20 00:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(16) "Europe/Amsterdam"
}
```

**Output for 5.4.30 - 5.4.35, 5.5.14 - 5.6.3, php7@20140701 - 20141101**
```
object(DateTime)#1 (3) {
  ["date"]=>
  string(26) "2015-01-01 00:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(16) "Europe/Amsterdam"
}
```
http://3v4l.org/9b0JY

* DateTime Relative Formats Reference: http://php.net/manual/en/datetime.formats.relative.php

Same issue with the keyword `last day of`

**Comment by Cholik** [comment]
Just figured out that following works as it should:

```php
$firstDay=date('Y-m-d',strtotime("first day of this month"));
var_dump($firstDay)
```

**Comment by paulbiss** [comment]
This looks very similar to #958

**Comment by jacobsantos** [comment]
@paulbiss Irrelevant. Problem persists. The other was closed for a specific problem, either regression happened or use case was not tested and fixed.

I'm not saying HHVM needs to match PHP5 or PHP7 exactly, however, it does need to give correct results. It would be better, if a more exhaustive test case was created for all of the relative terms and with more than a few use cases.

**Comment by Rattler3** [comment]
This is still an issue in the HHVM-3.7.1 release.

```php
$start = new \DateTime('2015-05-10 10:00:00');
$start->modify('first day of this month')->setTime(0,0,0);
echo $start->format('c')."\n";

$end = new \DateTime('2015-05-10 10:00:00');
$end->modify('last day of this month')->setTime(23,59,59);
echo $end->format('c')."\n";
```

PHP output 
```
2015-05-01T00:00:00+01:00
2015-05-31T23:59:59+01:00
```
HHVM output
```
2015-05-10T00:00:00+01:00
2015-05-10T23:59:59+01:00
```

**Comment by unframework** [comment]
Observed on 3.6.5:

```
vagrant@dev:~$ hhvm --version
HipHop VM 3.6.5 (rel)
Compiler: tags/HHVM-3.6.5-0-g20a30678cd67fad96602ffd93e69780d001ce57f
Repo schema: 53a4026d3732c3584cffef19fa47fea655be3c4f
```

**Comment by jazzdan** [comment]
This may be fixed now that https://github.com/hhvm/hhvm-third-party/pull/71 has landed in master.

Can you try a nightly or a master build and see if the problem persists?

**Comment by unframework** [comment]
Apologies, I missed the fact that it was fixed already. We will try a workaround while the fix makes its way downstream.

**Comment by fredemmott** [comment]
Closing as fixed in 3.8. The fix has not been backported to 3.3, 3.6, or 3.7.

**Comment by fredemmott** [comment]
Fix was https://github.com/facebook/hhvm/commit/e143c85c45f98834751e263ace814b3741eb6c5f
